lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2792,17 +2817,34 @@ bool TypeSystemSwiftTypeRef::DumpTypeValue(
   };
 
 #ifndef NDEBUG
+  auto result = impl();
+  if (!result) {
+    if (!m_swift_ast_context)
+      return false;
+    // Swift reflection provided no result, fallback to the AST.
+    return m_swift_ast_context->DumpTypeValue(
+        ReconstructType(type), s, format, data, data_offset, data_byte_size,
+        bitfield_bit_size, bitfield_bit_offset, exe_scope, is_base_class);
```

Comment:
The `impl` returns an `Optional`. When the optional is none, then this code falls back to using `SwiftASTContext`. This happens for enums only. Specifically these two cases:

1. Multipayload enums not supported by Swift reflection
2. Some clang imported enums

Reply:
That sounds good!

Reply:
> The impl returns an Optional. When the optional is none, then this code falls back to using SwiftASTContext. This happens for enums only. Specifically these two cases

It might help to put this as a comment in the code.

About the `Optional<bool>`: Could we replace the `return {}` with `return m_swift_ast_context->DumpTypeValue(
         ReconstructType(type), s, format, data, data_offset, data_byte_size,
         bitfield_bit_size, bitfield_bit_offset, exe_scope, is_base_class);` and then still use the `VALIDATE_AND_RETURN` macro?

Reply:
We could replace it. In that case the validation would be tautological, which I was trying to avoid. But it would make for much simpler code to do it that way. Now that we're discussing it, I guess it's better to make a tradeoff of simpler code at the cost of some cases of validation being useless?

Reply:
Yeah the validation would compare against itself and re-execute the same code, but the flow would be identical to all other functions thus the code might be easier to read/understand/maintain. Just my personal preference, no big deal :-)

Reply:
I think it's a good point

Reply:
done